src/components/Widget.tsx

```diff
@@ -93,10 +93,11 @@ export type WidgetProps = {
   dialog?: HTMLElement | null
   className?: string
   onError?: ErrorHandler
+  onConnect?: (provider: JsonRpcProvider) => void
```

Comment:
is this supposed to be `onConnectWallet`? 
![image](https://user-images.githubusercontent.com/5773490/168675179-ee1f56e5-b065-41ad-9db9-18ef9574a490.png)

Reply:
https://docs.uniswap.org/sdk/widgets/swap-widget/api

Reply:
This is a separate callback that I added to facilitate sequencing in testing. `onConnectWallet` is triggered when a user attempts to connect a wallet through pressing the "Connect your wallet" button; this is triggered when any provider is successfully connected.

Reply:
I really don't like adding undocumented API surface explicitly for the purpose of testing. If possible (imo) we should only test using the attributes naturally available to integrators and users. 

In this case, I'd make an assertion about expected UI state instead of adding a new prop to test.